homeassistant/components/intelliclima/fan.py

```diff
@@ -0,0 +1,250 @@
+"""Fan platform for IntelliClima VMC."""
+
+from dataclasses import dataclass
+import math
+from typing import Any
+
+from pyintelliclima import IntelliClimaECO
+
+from homeassistant.components.fan import (
+    FanEntity,
+    FanEntityDescription,
+    FanEntityFeature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+from homeassistant.util.scaling import int_states_in_range
+
+from .const import (
+    FAN_MODE_ALTERNATE,
+    FAN_MODE_IN,
+    FAN_MODE_OFF,
+    FAN_MODE_OUT,
+    FAN_MODE_SENSOR,
+    FAN_SPEED_AUTO,
+    FAN_SPEED_HIGH,
+    FAN_SPEED_OFF,
+    FAN_SPEED_SLEEP,
+)
+from .coordinator import IntelliClimaConfigEntry, IntelliClimaCoordinator
+from .entity import IntelliClimaEntity
+
+PRESET_MODES_TO_INTELLICLIMA_MODE = {
+    "off": FAN_MODE_OFF,
+    "forward": FAN_MODE_IN,
+    "reverse": FAN_MODE_OUT,
+    "alternate": FAN_MODE_ALTERNATE,
+    "sensor": FAN_MODE_SENSOR,
+    "auto": FAN_MODE_SENSOR,
+}
+INTELLICLIMA_MODE_TO_PRESET_MODES = {
+    v: k for k, v in PRESET_MODES_TO_INTELLICLIMA_MODE.items() if k != "auto"
+}
+
+
+@dataclass(frozen=True)
+class IntelliClimaFanRequiredKeysMixin:
+    """Required keys for fan entity."""
+
+    speed_range: tuple[int, int]
+
+
+@dataclass(frozen=True)
+class IntelliClimaFanEntityDescription(
+    FanEntityDescription, IntelliClimaFanRequiredKeysMixin
+):
+    """Describes a fan entity."""
+
+
+INTELLICLIMA_FANS: tuple[IntelliClimaFanEntityDescription, ...] = (
+    IntelliClimaFanEntityDescription(
+        key="fan",
+        translation_key="fan",
+        speed_range=(int(FAN_SPEED_SLEEP), int(FAN_SPEED_HIGH)),
```

Comment:
Is there a reason this is here now and not as an `IntelliClimaVMCFan` class attribute/property? Do you have plan to support other fan types?

Reply:
Well I cannot easiliy implement other fan types myself as it would require reverse engineering the api for those and you need to own those units. I only own the ECOCOMFORT 2.0, but others from FantiniCosmi do exist so I'm keeping this here so others can add units if they own them.

Reply:
done